src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -181,66 +174,59 @@ function addScriptTagsToDocument(resourceLoader: WebAssemblyResourceLoader) {
-  if (resourceLoader.bootConfig.cacheBootResources) {
-    scriptElem.integrity = dotnetJsContentHash;
-    scriptElem.crossOrigin = 'anonymous';
-  }
+  let src = `./${dotnetJsResourceName}`;
+
+  // TODO How to do SRI dotnetJsContentHash with dynamic import ?
 
   // Allow overriding the URI from which the dotnet.*.js file is loaded
   if (resourceLoader.startOptions.loadBootResource) {
     const resourceType: WebAssemblyBootResourceType = 'dotnetjs';
-    const customSrc = resourceLoader.startOptions.loadBootResource(resourceType, dotnetJsResourceName, scriptElem.src, dotnetJsContentHash);
+    const customSrc = resourceLoader.startOptions.loadBootResource(resourceType, dotnetJsResourceName, src, dotnetJsContentHash);
     if (typeof (customSrc) === 'string') {
-      scriptElem.src = customSrc;
+      src = customSrc;
     } else if (customSrc) {
       // Since we must load this via a <script> tag, it's only valid to supply a URI (and not a Request, say)
       throw new Error(`For a ${resourceType} resource, custom loaders must supply a URI string.`);
     }
   }
 
-  document.body.appendChild(scriptElem);
+  return new Promise((resolve, reject) => {
+    // GOTCHA: remove this callback once we switch to ES6
+    // this is callback we have in CJS version of the runtime
+    globalThis.__onDotnetRuntimeLoaded = (createDotnetRuntime) => {
+      delete globalThis.__onDotnetRuntimeLoaded;
+      resolve(createDotnetRuntime);
+    };
+    // this tells webpack to do dynamic import at runtime, not at compile time
+    import(/* webpackIgnore: true */ src).then(({ default: createDotnetRuntime }) => {
+      if (createDotnetRuntime) {
+        // this runs when loaded module was ES6
+        delete globalThis.__onDotnetRuntimeLoaded;
+        resolve(createDotnetRuntime);
+      }
+    }, reject);
+  });
 }
 
-// Due to a strange behavior in macOS Catalina, we have to delay loading the WebAssembly files
-// until after it finishes evaluating a <script> element that assigns a value to window.Module.
-// This may be fixed in a later version of macOS/iOS, or even if not it may be possible to reduce
-// this to a smaller workaround.
-function addGlobalModuleScriptTagsToDocument(callback: () => void) {
-  const scriptElem = document.createElement('script');
-
-  // This pollutes global but is needed so it can be called from the script.
-  // The callback is put in the global scope so that it can be run after the script is loaded.
-  // onload cannot be used in this case for non-file scripts.
-  window['__wasmmodulecallback__'] = callback;
+async function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourceLoader, onReady: () => void, onError: (reason?: any) => void) {
 
-  // Note: Any updates to the following script will require updating the inline script hash if using CSP
-  scriptElem.text = 'var Module; window.__wasmmodulecallback__(); delete window.__wasmmodulecallback__;';
```

Comment:
🎉  this removal is great! Fixes: https://github.com/dotnet/aspnetcore/issues/36805

Did you get a chance to validate it using MacOS Catalina?

Reply:
Nope, I don't have access to that device.

Reply:
Ah okay, let me give it a try...

Reply:
Just tried it on my Mac (running Big Sur -> successor to Catalina). Didn't encounter any issues. I don't have a Catalina device to verify on. 

Given `This may be fixed in a later version of macOS/iOS`, I'm going to consider this resolved now.